### PR TITLE
SALTO-5716: Fixing spelling errors in SFDC adapter

### PR DESCRIPTION
**Pull request by yelly**
Some typos fixed, some changes just to make the spellcheck happy.

---

_Additional context for reviewer_

---
_Release Notes_: 
None.

---
_User Notifications_: 
None.

### CODE REVIEW

**Review by tomsellek — COMMENTED** [review]
Boyscouting FTW!!!

There are a couple of places where I think it would be better to stick to Salesforce terminology, but like 95% of this change would just make our lives easier in subtle ways. Thanks!

Also, do you think it would make sense to add this commit rev. to `.git-blame-ignore-revs`? I'm kinda on the fence myself.

File: packages/salesforce-adapter/test/filters/custom_objects_instances.test.ts [hunk 1]
```diff
@@ -167,7 +167,7 @@ describe('Custom Object Instances filter', () => {
       Id: 'abcdefg',
       OtherAddress: {
         city: 'city',
-        counry: 'country',
+        country: 'country',
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/test/filters/custom_objects_instances.test.ts:170]: IIUC this is a custom object that is defined by the test, so the content doesn't actually matter.

File: packages/salesforce-adapter/test/transformers/transformer.test.ts [hunk 2]
```diff
@@ -1176,7 +1176,7 @@ describe('transformer', () => {
         expect(_.get(customObj, DESCRIPTION)).toEqual('MyDescription')
       })
 
-      it('should not transform skiplisted annotations', () => {
+      it('should not transform skip listed annotations', () => {
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/test/transformers/transformer.test.ts:1179]: It actually isn't, the variable name is `annotaionsToSkip`, so talking about "skipped annotations" is accurate here.

File: packages/salesforce-adapter/test/change_validators/full_name_change.test.ts [hunk 3]
```diff
@@ -14,47 +14,47 @@
  * limitations under the License.
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/test/change_validators/full_name_change.test.ts:14]: `fullName` in snake-case is spelled `full_name`.

File: packages/salesforce-adapter/test/change_validators/animation_rule_record_type.test.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ import {
   ObjectType,
   toChange,
 } from '@salto-io/adapter-api'
-import changeValidator from '../../src/change_validators/animation_rule_recordtype'
+import changeValidator from '../../src/change_validators/animation_rule_record_type'
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/test/change_validators/animation_rule_record_type.test.ts:22]: `RecoredType` in snake case is spelled `record_type`.

**Comment by yelly** [comment]
> Also, do you think it would make sense to add this commit rev. to `.git-blame-ignore-revs`? I'm kinda on the fence myself.

I think not, this commit isn't really big enough.

**Review by tomsellek — APPROVED** [review]
Love it

**Pull request merged**